pkg/controller/installation/windows_controller.go

```diff
@@ -0,0 +1,597 @@
+		}
+
+		nodePrometheusTLS, err = certificateManager.GetOrCreateKeyPair(r.client, render.NodePrometheusTLSServerSecret, common.OperatorNamespace(), dns.GetServiceDNSNames(render.CalicoNodeMetricsService, common.CalicoNamespace, r.clusterDomain))
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceCreateError, "Error creating TLS certificate", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		if nodePrometheusTLS != nil {
+			typhaNodeTLS.TrustedBundle.AddCertificates(nodePrometheusTLS)
+		}
+		prometheusClientCert, err := certificateManager.GetCertificate(r.client, monitor.PrometheusClientTLSSecretName, common.OperatorNamespace())
+		if err != nil {
+			r.status.SetDegraded(operatorv1.CertificateError, "Unable to fetch prometheus certificate", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		if prometheusClientCert != nil {
+			typhaNodeTLS.TrustedBundle.AddCertificates(prometheusClientCert)
+		}
+	}
+
+	// Secure calico kube controller metrics.
+	var kubeControllerTLS certificatemanagement.KeyPairInterface
+	if instance.Spec.Variant == operatorv1.TigeraSecureEnterprise {
+		// Create or Get TLS certificates for kube controller.
+		kubeControllerTLS, err = certificateManager.GetOrCreateKeyPair(
+			r.client,
+			kubecontrollers.KubeControllerPrometheusTLSSecret,
+			common.OperatorNamespace(),
+			dns.GetServiceDNSNames(kubecontrollers.KubeControllerMetrics, common.CalicoNamespace, r.clusterDomain))
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Error finding or creating TLS certificate kube controllers metric", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+
+		// Add prometheus client certificate to Trusted bundle.
+		kubecontrollerprometheusTLS, err := certificateManager.GetCertificate(r.client, monitor.PrometheusClientTLSSecretName, common.OperatorNamespace())
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get certificate for kube controllers", err, reqLogger)
+			return reconcile.Result{}, err
+		} else if kubecontrollerprometheusTLS != nil {
+			typhaNodeTLS.TrustedBundle.AddCertificates(kubeControllerTLS, kubecontrollerprometheusTLS)
+		}
+	}
+
+	components := []render.Component{}
+
+	// Fetch any existing default BGPConfiguration object.
+	bgpConfiguration := &crdv1.BGPConfiguration{}
+	err = r.client.Get(ctx, types.NamespacedName{Name: "default"}, bgpConfiguration)
+	if err != nil && !apierrors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Unable to read BGPConfiguration", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	var hasWindowsNodes bool
+	// Don't render calico-node-windows if it's disabled in the installation
+	if instance.Spec.CalicoNetwork != nil && (instance.Spec.CalicoNetwork.WindowsDataplane == nil || *instance.Spec.CalicoNetwork.WindowsDataplane == operatorv1.WindowsDataplaneDisabled) {
+		reqLogger.Info("Calico Windows daemonset is disabled in the operator installation")
```

Comment:
Instead of just logging, we should `return reconcile.Result{}, nil` - that way we simply stop execution of the controller, and we don't need an "else" branch to this "if"